backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -1295,10 +1295,50 @@ private boolean checkAllResourceAllowed(List<Resources> resources) {
   }
 
   public InternalFuture<ExperimentRun> createExperimentRun(CreateExperimentRun request) {
-    return getEntityPermissionBasedOnResourceTypes(
-            Collections.singletonList(request.getProjectId()),
-            ModelDBActionEnum.ModelDBServiceActions.UPDATE,
-            ModelDBServiceResourceTypes.PROJECT)
+    // Validate arguments
+    var futureTask =
+        InternalFuture.runAsync(
+            () -> {
+              if (request.getProjectId().isEmpty()) {
+                throw new InvalidArgumentException(
+                    "Project ID not found in CreateExperimentRun request");
+              } else if (request.getExperimentId().isEmpty()) {
+                throw new InvalidArgumentException(
+                    "Experiment ID not found in CreateExperimentRun request");
+              }
+            },
+            executor);
+    return futureTask
+        .thenCompose(
+            unused ->
+                getEntityPermissionBasedOnResourceTypes(
+                    Collections.singletonList(request.getProjectId()),
+                    ModelDBActionEnum.ModelDBServiceActions.UPDATE,
+                    ModelDBServiceResourceTypes.PROJECT),
+            executor)
+        .thenAccept(
+            allowed -> {
+              if (!allowed) {
+                throw new PermissionDeniedException("Permission denied");
+              }
+            },
+            executor)
+        .thenCompose(
+            unused ->
+                jdbi.useHandle(
+                    handle -> {
+                      Long count =
+                          handle
+                              .createQuery("SELECT COUNT(*) FROM experiment where id = :id")
```

Comment:
```suggestion
                              .createQuery("SELECT COUNT(id) FROM experiment where id = :id")
```
Don't `count(*)` ever. Not every DB knows how to do it efficiently. Instead, count an indexed value.

Reply:
Fixed. it was my bad